src/libs/scene_utils/client/cesium.ts

```diff
@@ -0,0 +1,400 @@
+    });
+  }
+
+  private async handleCesiumAPI(url: string, options: RequestInit, errorMessage: (res: Response) => string): Promise<any> {
+    const response = await this.handleRequest(url, { ...options, compress: false }, errorMessage);
+    return this.handleResponse(response.body);
+  }
+
+  // Tell Cesium we want to upload a file
+  private async uploadAssetRequest(accessToken: string, request: UploadAssetRequest) {
+    const url = 'https://api.cesium.com/v1/assets';
+    const body = JSON.stringify(request);
+    const options: RequestInit = {
+      method: 'POST',
+      body,
+      headers: { 
+        'Content-Type': 'application/json',
+        'Content-Length': body.length.toString(),
+        Authorization: `Bearer ${accessToken}`,  
+      },
+    };
+    const errorMessage = (res: Response) => {
+      return `[Error ${res.status}] Failed to process asset upload request with error: ${res.statusText}`;
+    }
+
+    return this.handleCesiumAPI(url, options, errorMessage);
+  }
+
+  // Upload file to S3 location provided by Cesium
+  private async uploadAssetToS3Location(assetPath: string, uploadLocation: UploadLocation) {
+    const cesiumS3 = new S3({
+      region: 'us-east-1',
+      signatureVersion: 'v4',
+      endpoint: uploadLocation.endpoint,
+      credentials: new Credentials(
+        uploadLocation.accessKey,
+        uploadLocation.secretAccessKey,
+        uploadLocation.sessionToken,
+      ),
+    });
+
+    const fileName = basename(assetPath);
+    return cesiumS3
+      .upload({
+        Body: createReadStream(assetPath),
+        Bucket: uploadLocation.bucket,
+        Key: `${uploadLocation.prefix}${fileName}`,
+      })
+      .promise();
+  }
+
+  // Tell Cesium the file has been uploaded to S3 and start tiling
+  private async startTilingForAsset(accessToken: string, onComplete: OnComplete) {
+    const url = onComplete.url;
+    const options: RequestInit = {
+      method: onComplete.method,
+      body: JSON.stringify(onComplete.fields),
+      headers: { Authorization: `Bearer ${accessToken}` },
+    };
+    const errorMessage = (res: Response) => {
+      return `[Error ${res.status}] Failed to request tiling on Cesium asset with error: ${res.statusText}`;
+    }
+
+    return this.handleCesiumAPI(url, options, errorMessage);
+  }
+
+  // Log progress of tiling for a Cesium asset
+  private async waitForTiles(
+    accessToken: string,
+    assetId: string,
+    timeoutInMs: number = 10000, // Default refresh every 10 seconds
+    maxChecks: number = 30, // Default wait for 5 minutes
+  ) {
+    let done = false;
+
+    while (!done && maxChecks-- > 0) {
+      // Issue a GET request for the metadata
+      const response = await this.getAsset(accessToken, assetId);
+      const assetMetadata = JSON.parse(response.toString());
+      const status = assetMetadata.status;
+      if (status === 'COMPLETE') {
+        process.stdout.clearLine(0);
+        console.log('\nAsset tiled successfully');
```

Comment:
You can consolidate these into a single `console.log()` function with comma delineation. 
`console.log('\nAsset tiled successfully', 'View in Cesium Ion: https://cesium.com/ion/assets/${assetId}')`